File: solutions/_1758_minimum_changes_to_make_alternating_binary_string/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function (s) {
  let count1 = 0;
  let count2 = 0;
  for (let i = 0; i < s.length; i++) {
    if (s[i] == i % 2) {
      count1++;
    } else {
      count2++;
    }
  }
  return Math.min(count1, count2);
};

module.exports = { minOperations };

/*
https://leetcode.com/problems/minimum-changes-to-make-alternating-binary-string/
*/

File: solutions/_1758_minimum_changes_to_make_alternating_binary_string/solution.java
package _1758_minimum_changes_to_make_alternating_binary_string;

class Solution {
    public int minOperations(String s) {
        int n = s.length();
        int count1 = 0;
        int count2 = 0;
        for (int i = 0; i < n; i++) {
            if (s.charAt(i) - '0' == i % 2)
                count1++;
            else
                count2++;
        }
        return Math.min(count1, count2);
    }
}

/**
 * https://leetcode.com/problems/minimum-changes-to-make-alternating-binary-string/
 */